linker/descriptors.go

```diff
@@ -400,13 +400,13 @@ func (s *srcLocs) ByDescriptor(d protoreflect.Descriptor) protoreflect.SourceLoc
 
 type msgDescriptors struct {
 	protoreflect.MessageDescriptors
-	msgs []*msgDescriptor
+	msgs []msgDescriptor
```

Comment:
For simplicity could you see what the performance is of using the flattened slices without the allocPool? It looks like it should be equivalent for `msgDescriptors` if `createMessages` is called once per file descriptor.

Reply:
My first pass at this basically looked like that -- each `msgDescriptors` instance or `fldDescriptors`, etc, only allocated its own flattened slice. It was a small improvement over the baseline, but this was actually a larger improvement, primarily because fields are a much larger source of allocations than messages.

I don't think it makes sense to handle the different descriptor types differently in here -- like a hybrid where some types just allocate a slice and others use the pool. That would be strangely inconsistent, which IMO would not be simpler.

If you're really curious to see the benchmark output for the various states, I should have the original commits in my local repo (recoverable via `git reflog`). This one commit was originally three, so I had three different benchmark runs. All of the changes were incremental and had observable but relatively minor impacts on the benchmarks. (Next PR, #291, had a much bigger impact.)

Reply:
You can see the details over here: #292. This shows the actual incremental steps I took to arrive at this branch, as well as the (admittedly very small) incremental improvements each step had on the benchmark I was running.